internal/api/v1/user/totp/errors.go

```diff
@@ -0,0 +1,10 @@
+package totp
+
+import "errors"
+
+var (
+	// ErrMissingXpriv is returned when the xpriv is missing.
+	ErrMissingXpriv = errors.New("xpriv is missing")
+	// ErrContactPubKeyInvalid is returned when the contact's PubKey is invalid.
+	ErrContactPubKeyInvalid = errors.New("contact's PubKey is invalid")
+)
```

Comment:
this is the discussion per errors, where u said this:  
> The concept is to define errors within the package from which they are returned. Plus, use what the standard library offers for all errors that are not returned after sending a request to the SPV Wallet API. Take a look at the example in the tests of one of the packages, e.g., transactions_test.

Note that the errors you pasted are defined within the internal package, which no one from the outside will have access to. So, they won't be able to handle them, e.g., use errors.Is(), etc.

Reply:
Exactly, and the last part of the text highlights a visible problem with external error handling or checking. That's why I suggested extracting these errors to the wallet package for improved accessibility and proper error management."

@chris-4chain @dorzepowski  what do you think about this?

Reply:
so why before we defining errors in internal package ? you asked to follow of what you did if i correctly understood you .. now we wanna make them as separate errors and be public ?

Reply:
Yes, I agree. In my opinion, the previous approach of storing the errors in the internal package is more of a design issue that can be easily addressed.

In internal packages, we're not handling errors in a specific way; it's generally the caller's responsibility. However, as I mentioned, when the client returns an error like `MetadataDepthExceeded`, the caller won't be able to catch that failure using the errors package API, which is a problem.

To avoid mixing these fixes with PRs, it would be better to create a separate task and fix it easily :)